ledfx/effects/bar.py

```diff
@@ -49,13 +52,55 @@ def on_activate(self, pixel_count):
         self.color_idx = 0
         self.bar_len = 0.3
         self.beat_count = 0
+        self.beat_count_before = 0
+        self.delayed = False
+        self.first_beat = True
+        self.has_beat_once = False
+        self.beats_since_active = 0
         self.last_beat = 0.0
 
     def audio_data_updated(self, data):
         # Run linear beat oscillator through easing method
         self.beat_oscillator = data.beat_oscillator()
         self.beat_now = data.bpm_beat_now()
-        self.beat_count = data.beat_counter
+
+        # Skips first beat after activation to sync up all segments
+        if self.first_beat:
+            self.beat_count_before = data.beat_counter
+            self.beat_count = data.beat_counter
+            self.first_beat = False
+        else:
+            self.beat_count_before = self.beat_count
+            self.beat_count = data.beat_counter
+
+        # Fixes a bug where delay would be out of sync when music is not playing on effect activation
+        if not self.has_beat_once:
+            if self.beat_count_before != self.beat_count:
+                self.has_beat_once = True
+                return
+            else:
+                return
+
+        # Counts the beats since activation of effect; Unblocks render function when specified amount of beats is reached
+        if self._config["delay_beats"] > 0 and not self.delayed:
+            print(
```

Comment:
only use _LOGGER for debug, assuming you would delete this anyway before submission, ie if its just use shaking code, but get used to using LOGGER anyway

I can point at examples if needed.